docusaurus/docs/iOS/components/message-list.md

```diff
@@ -1,3 +1,272 @@
 ---
 title: Message List
----
\ No newline at end of file
+---
+
+import ComponentsNote from '../common-content/components-note.md'
+import MessageListProperties from '../common-content/reference-docs/stream-chat-ui/chat-message-list/chat-message-list-vc-properties.md'
+
+The `ChatMessageListVC` is a component responsible to display a list of messages. It is not the responsibility of the message list to know the source of the messages, the data should be provided through the `ChatMessageListVCDataSource` and the `ChatMessageListVCDelegate`, very similar to how a native `UITableView` and a `UICollectionView` works.
+
+:::note
+The Stream SDK already provides a `ChatChannelVC` and a `ChatThreadVC` that use the `ChatMessageListVC` to render the messages from a Channel and Thread, respectively.
+:::
+
+## Usage
+
+In case you want to compose a custom Chat view and not use our built-in `ChatChannelVC` and `ChatThreadVC` components, you can use `ChatMessageListVC` to display the list of messages. The messages should come from a `ChannelController` or a `MessageController` depending if you want to display messages from a channel or a thread.
+
+First, we need to setup our view controller and set the layout constraints of the message list. For simplicity, we will be using the `ComposerVC` included in the Stream SDK, but you could provide your own message composer as well.
+```swift
+open class ChannelViewController: UIViewController, ThemeProvider {
+    /// Controller for observing data changes within the channel.
+    open var channelController: ChatChannelController!
+
+    /// User search controller for suggestion users when typing in the composer.
+    open lazy var userSuggestionSearchController: ChatUserSearchController =
+        channelController.client.userSearchController()
+
+    /// The message list component responsible to render the messages.
+    open lazy var messageListVC: ChatMessageListVC = ChatMessageListVC()
+
+    /// Controller that handles the composer view.
+    open lazy var messageComposerVC = ComposerVC()
+
+    /// Flag to control if it is currently loading previous messages.
+    @Atomic private var loadingPreviousMessages: Bool = false
```

Comment:
can we move the `loadingPreviousMessages` inside the framework? We should not leave this to the customer

Reply:
Yes I was discussing with @kojiba to move this to the LLC

Reply:
In the documentation I can remove this for now, and just call:
```
channelController.loadPreviousMessages()
```